tests/python/pants_test/util/test_objects_mypy.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
Do you think we need this test? MyPy tests pretty intensely the dataclass plugin already. The part that we need to make sure we get right is the interaction between engine and dataclasses, which you do a good job testing in `test_scheduler.py`.